### ISSUE TITLE
java cachedHostConnectionPool API could not setting pool by programming

### ISSUE THREAD

**Issue by XuefengWu** [akka/akka-http#857]
In safety internal network environment,  some system using http and have host-level limitation.
There is lack API for setting HTTP host-level connection pool. 

The exists create pool with settings API is only for HTTPS.
[javadsl/Http.scala](https://github.com/akka/akka-http/blob/master/akka-http-core/src/main/scala/akka/http/javadsl/Http.scala)
```
  /**
   * Same as [[cachedHostConnectionPool]] but with HTTPS encryption.
   *
   * The given [[ConnectionContext]] will be used for encryption on the connection.
   */
  def cachedHostConnectionPool[T](
    to:       ConnectHttp,
    settings: ConnectionPoolSettings,
    log:      LoggingAdapter, materializer: Materializer): Flow[Pair[HttpRequest, T], Pair[Try[HttpResponse], T], HostConnectionPool] =
    adaptTupleFlow(delegate.cachedHostConnectionPoolHttps[T](to.host, to.port, to.effectiveHttpsConnectionContext(defaultClientHttpsContext).asScala, settings.asScala, log)(materializer)
      .mapMaterializedValue(_.toJava))
```

t:http:core

**Comment by jrudolph** [comment]
This is the most general API entry-point which should allow both for HTTP and HTTPS connection pools. It seems that just the javadoc is wrong.

**Comment by jrudolph** [comment]
On the other hand there's also this one:

```scala
  def cachedHostConnectionPool[T](to: ConnectHttp, materializer: Materializer): Flow[Pair[HttpRequest, T], Pair[Try[HttpResponse], T], HostConnectionPool] =
    adaptTupleFlow(delegate.cachedHostConnectionPool[T](to.host, to.port)(materializer).mapMaterializedValue(_.toJava))
```

which looks similarly as if an https connection pool could be created with that (using `ConnectHttp`) but which isn't possible. I think these might need to be revised.

**Comment by laszlovandenhoek** [comment]
In the current 10.1.3, using the Java DSL, it is impossible to use custom `ConnectionPoolSettings` with a plain (non-HTTPS) `cachedHostConnectionPool`. The only overload that allows specifying it, is the deprecated 4-arg version, and using it doesn't yield the desired result; since it delegates internally to `cachedHostConnectionPoolHttps`, it forces all connections going through it to be treated as HTTPS, which of course causes requests to fail because of a missing SSL handshake.

My workaround has been to create a wrapper that exposes a Java interface, calling the Scala DSL version of `cachedHostConnectionPool`, which (correctly) delegates to `cachedHostConnectionPoolImpl`, leaving requests as plain HTTP.

**Comment by jrudolph** [comment]
Fixed by #2562.
